### PR TITLE
desktop-file-utils: wrap update-desktop-database to make icon catalog launchers

### PR DESCRIPTION
**Pull request by mach-kernel**
#141 

- At first I used `iconcatalogedit`, which AFAICT just makes a symlink and calls it a day
- I got that working, only to find out that the apps would refuse to launch because of lack of SGUG env
  - `/usr/lib/X11/xdm/Xsession` breaks a bunch of desktop stuff (fonts, colors, etc) if started in SGUG env
- Looks like you can also directly place scripts in the icon catalog, which is what I settled for
- ~rpm manages the 'launcher' for a given package and cleanly uninstalls when gone, tested with `nedit`~
  - No longer necessary! Hooking on `update-desktop-database` was a much better bet
  - Makes sure there is parity between the `/usr/sgug/share/applications` and the catalog
- Decided to install to system-wide icon catalog under the `RSE` page (you can control this with the path)

TODO

- Icons (need to finish investigating)

### CODE REVIEW

File: packages/nedit/SPECS/nedit.spec [hunk 1]
```diff
@@ -77,6 +77,12 @@ desktop-file-install \
         --add-category "Development;" \
         %{SOURCE1}
 
+%pre
```
Comment (mach-kernel) [inline packages/nedit/SPECS/nedit.spec:80]: I'm going to try to wrap `desktop-file-install` again and try to run this after it and somehow stage it in `$BUILDROOT`

File: packages/nedit/SPECS/nedit.spec [hunk 2]
```diff
@@ -79,6 +79,7 @@ desktop-file-install \
 
 %files
 %doc README ReleaseNotes
+/usr/lib/desktop/iconcatalog/pages/C/RSE/*
```
Comment (mach-kernel) [inline packages/nedit/SPECS/nedit.spec:82]: This is the only change that we need to make to the specs for inclusion in the icon catalog. I don't know how we feel about the RPMs managing stuff outside of our prefix, _but_ uninstalling the RPM does also take the icon catalog entry with it by doing this!!

File: packages/desktop-file-utils/SOURCES/sgugenv.sh [hunk 3]
```diff
@@ -0,0 +1,52 @@
+#!/usr/sgug/bin/bash
```
Comment (mach-kernel) [inline packages/desktop-file-utils/SOURCES/sgugenv.sh:1]: #147 can help me here 😸

File: packages/desktop-file-utils/SOURCES/desktop-file-utils-wrapper-install.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+DFI_DESKTOP_KEY_EXEC=$DFI_LAST_KEY
+
+INSTALL_ROOT=$(dirname $DFI_DIR | sed 's,/usr/sgug/.*,,')
+CATALOG_ROOT=$INSTALL_ROOT/usr/lib/desktop/iconcatalog/pages/C/RSE
+
+mkdir -p $CATALOG_ROOT
+cp /usr/sgug/share/desktop-file-utils/sgugenv.sh $CATALOG_ROOT/$DFI_DESKTOP_KEY_EXEC
+
+chmod 755 $CATALOG_ROOT/$DFI_DESKTOP_KEY_EXEC
+echo "exec $DFI_DESKTOP_KEY_EXEC" >> $CATALOG_ROOT/$DFI_DESKTOP_KEY_EXEC
```
Comment (mach-kernel) [inline packages/desktop-file-utils/SOURCES/desktop-file-utils-wrapper-install.sh:40]: I'm sure this constitutes an abuse but I couldn't figure out a better way!

File: packages/desktop-file-utils/SOURCES/desktop-file-utils-wrapper-install.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+			shift 2
+			;;
+		*)
+			break
+			;;
+    esac
+done
+
+# We should only do this for an rpmbuild
+if [[ "$DFI_DIR" != *"BUILDROOT"* ]]; then
```
Comment (mach-kernel) [inline packages/desktop-file-utils/SOURCES/desktop-file-utils-wrapper-install.sh:21]: Is this 'safe' enough?

File: packages/desktop-file-utils/SOURCES/update-desktop-database-wrapper.sh [hunk 6]
```diff
@@ -35,6 +35,8 @@ for DESKTOP_FILE_PATH in $(find $SGUG_DATADIR_APPS/*.desktop -type f); do
 	cp /usr/sgug/share/desktop-file-utils/sgugenv.sh $CATALOG_ROOT/$APPNAME
 
 	chmod 755 $CATALOG_ROOT/$APPNAME
-	echo "exec $DESKTOP_KEY_EXEC" >> $CATALOG_ROOT/$APPNAME
+	echo "#!/usr/sgug/bin/bash" >> $CATALOG_ROOT/$APPNAME
+	echo "export PATH=$PATH:/usr/sgug/bin" >> $CATALOG_ROOT/$APPNAME
```
Comment (danielhams) [inline packages/desktop-file-utils/SOURCES/update-desktop-database-wrapper.sh:39]: I think maybe this should be `PATH=/usr/sgug/bin:$PATH` to be sure we hit SGUG binaries before system ones.

**Review by danielhams — COMMENTED** [review]
Unsure if you want this merged like this yet. I leave this just as feedback for now.

Cheers.

File: packages/desktop-file-utils/SOURCES/update-desktop-database-wrapper.sh (hunk 6 above)
Comment (mach-kernel) [reply packages/desktop-file-utils/SOURCES/update-desktop-database-wrapper.sh:39]: Yes!

Looks like I also forgot that `cp` in the spec.

**Comment by mach-kernel** [comment]
https://www.aquaphoenix.com/software/ftieditor/page6.html is a small FTI spec, and also has a Java editor for download

Merging now, but we maybe can generate FTIs from SVGs. The painful bit is that there seems to be no support for doing parametric curves like you can with SVG. Everything would have to be points.

**Pull request merged**